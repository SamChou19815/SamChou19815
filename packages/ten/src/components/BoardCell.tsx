import React, { ReactElement } from 'react';
import styles from './BoardCell.module.css';

type Props = {
  readonly tileStatus: number;
  readonly doesNeedHighlight: boolean;
  readonly onClick: () => void;
};

export default function BoardCell({ tileStatus, doesNeedHighlight, onClick }: Props): ReactElement {
  let backgroundColor: string;
  if (tileStatus === 1) {
    backgroundColor = 'black';
  } else if (tileStatus === -1) {
    backgroundColor = 'white';
  } else if (tileStatus === 0) {
    backgroundColor = '#DDDDDD';
  } else {
    throw new Error(`Bad tile status: ${tileStatus}!`);
  }
  const border = doesNeedHighlight ? '1px solid red' : '1px solid #CCC';
  const style = { backgroundColor, border };
  return <div role="presentation" className={styles.BoardCell} style={style} onClick={onClick} />;
}
