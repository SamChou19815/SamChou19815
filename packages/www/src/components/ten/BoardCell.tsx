import React from 'react';

type Props = {
  readonly tileStatus: number;
  readonly doesNeedHighlight: boolean;
  readonly onClick: () => void;
};

export default function BoardCell({ tileStatus, doesNeedHighlight, onClick }: Props): JSX.Element {
  let backgroundColor: string;
  if (tileStatus === 1) {
    backgroundColor = 'black';
  } else if (tileStatus === -1) {
    backgroundColor = 'white';
  } else if (tileStatus === 0) {
    backgroundColor = '#DDDDDD';
  } else {
    throw new Error(`Bad tile status: ${tileStatus}!`);
  }
  const border = doesNeedHighlight ? '1px solid red' : '1px solid #CCC';
  const style = { backgroundColor, border };
  return (
    <div
      role="presentation"
      className="m-0.5 h-7 w-7 cursor-pointer"
      style={style}
      onClick={onClick}
    />
  );
}
