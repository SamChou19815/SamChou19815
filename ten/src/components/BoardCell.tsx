import React, { ReactElement } from 'react';
import styles from './BoardCell.module.css';

type Props = {
  readonly tileStatus: number;
  readonly doesNeedHighlight: boolean;
  readonly onClick: () => void;
};

/**
 * The board cell to display.
 *
 * @param {number} tileStatus the status of the tile.
 * @param {boolean} doesNeedHighlight whether the tile needs highlight.
 * @param {function(): void} onClick handle on click.
 * @return {Node} the rendered node.
 * @constructor
 */
export default function BoardCell({ tileStatus, doesNeedHighlight, onClick }: Props): ReactElement {
  let backgroundColor: string;
  if (tileStatus === 1) {
    backgroundColor = 'black';
  } else if (tileStatus === -1) {
    backgroundColor = 'white';
  } else if (tileStatus === 0) {
    backgroundColor = '#DDDDDD';
  } else {
    throw new Error(`Bad tile status: ${tileStatus}!`);
  }
  const border = doesNeedHighlight ? '1px solid red' : '1px solid #CCC';
  const style = { backgroundColor, border };
  return <div role="presentation" className={styles.BoardCell} style={style} onClick={onClick} />;
}
