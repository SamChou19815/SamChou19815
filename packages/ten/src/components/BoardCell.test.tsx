import React from 'react';
import renderer from 'react-test-renderer';
import BoardCell from './BoardCell';

const dummyOnClick = () => {};

it('BoardCell(1, true) matches snapshot.', () => {
  const tree = renderer
    .create(<BoardCell tileStatus={1} doesNeedHighlight onClick={dummyOnClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('BoardCell(1, false) matches snapshot.', () => {
  const tree = renderer
    .create(<BoardCell tileStatus={1} doesNeedHighlight={false} onClick={dummyOnClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('BoardCell(-1, true) matches snapshot.', () => {
  const tree = renderer
    .create(<BoardCell tileStatus={-1} doesNeedHighlight onClick={dummyOnClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('BoardCell(-1, false) matches snapshot.', () => {
  const tree = renderer
    .create(<BoardCell tileStatus={-1} doesNeedHighlight={false} onClick={dummyOnClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('BoardCell(0, true) matches snapshot.', () => {
  const tree = renderer
    .create(<BoardCell tileStatus={0} doesNeedHighlight onClick={dummyOnClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('BoardCell(0, false) matches snapshot.', () => {
  const tree = renderer
    .create(<BoardCell tileStatus={0} doesNeedHighlight={false} onClick={dummyOnClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
