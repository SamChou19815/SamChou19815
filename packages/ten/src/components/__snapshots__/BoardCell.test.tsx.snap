// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardCell(-1, false) matches snapshot. 1`] = `
<div
  className="BoardCell"
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "backgroundColor": "white",
      "border": "1px solid #CCC",
    }
  }
/>
`;

exports[`BoardCell(-1, true) matches snapshot. 1`] = `
<div
  className="BoardCell"
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "backgroundColor": "white",
      "border": "1px solid red",
    }
  }
/>
`;

exports[`BoardCell(0, false) matches snapshot. 1`] = `
<div
  className="BoardCell"
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "backgroundColor": "#DDDDDD",
      "border": "1px solid #CCC",
    }
  }
/>
`;

exports[`BoardCell(0, true) matches snapshot. 1`] = `
<div
  className="BoardCell"
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "backgroundColor": "#DDDDDD",
      "border": "1px solid red",
    }
  }
/>
`;

exports[`BoardCell(1, false) matches snapshot. 1`] = `
<div
  className="BoardCell"
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid #CCC",
    }
  }
/>
`;

exports[`BoardCell(1, true) matches snapshot. 1`] = `
<div
  className="BoardCell"
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid red",
    }
  }
/>
`;
